From 99b5e4b143e391991cfe06eebe34488afa9b9662 Mon Sep 17 00:00:00 2001
From: Yang Jihong <yangjihong1@huawei.com>
Date: Fri, 10 Mar 2023 11:43:57 +0800
Subject: [PATCH 354/921] [Huawei] bpf: fix KABI breakage in struct
 bpf_prog_aux for backporting kfunc_call patches

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: N/A

--------------------------------

Commit b0ae596cf225 ("[Backport] bpf: Support bpf program calling kernel
function") add new member in 'struct bpf_prog_aux', need to fix KABI
breakage.

Signed-off-by: Yang Jihong <yangjihong1@huawei.com>
---
 include/linux/bpf.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/bpf.h b/include/linux/bpf.h
index 2714c87c1df3..ff0a53ace3ec 100644
--- a/include/linux/bpf.h
+++ b/include/linux/bpf.h
@@ -21,6 +21,7 @@
 #include <linux/kallsyms.h>
 #include <linux/capability.h>
 #include <linux/percpu-refcount.h>
+#include <linux/kabi.h>
 
 struct bpf_verifier_env;
 struct bpf_verifier_log;
@@ -880,7 +881,6 @@ struct bpf_prog_aux {
 	struct bpf_prog **func;
 	void *jit_data; /* JIT specific data. arch dependent */
 	struct bpf_jit_poke_descriptor *poke_tab;
-	struct bpf_kfunc_desc_tab *kfunc_tab;
 	u32 size_poke_tab;
 	struct bpf_ksym ksym;
 	const struct bpf_prog_ops *ops;
@@ -927,6 +927,7 @@ struct bpf_prog_aux {
 		struct work_struct work;
 		struct rcu_head	rcu;
 	};
+	KABI_EXTEND(struct bpf_kfunc_desc_tab *kfunc_tab)
 };
 
 struct bpf_array_aux {
-- 
2.17.1

